src/VisualStudio/Core/Def/TaskList/ExternalErrorDiagnosticUpdateSource.cs

```diff
@@ -5,132 +5,103 @@
 using System;
 using System.Collections.Generic;
 using System.Collections.Immutable;
+using System.ComponentModel.Composition;
 using System.Diagnostics;
 using System.Linq;
 using System.Threading;
 using System.Threading.Tasks;
 using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.Collections;
 using Microsoft.CodeAnalysis.Diagnostics;
 using Microsoft.CodeAnalysis.Editor.Shared.Utilities;
-using Microsoft.CodeAnalysis.Internal.Log;
-using Microsoft.CodeAnalysis.Notification;
+using Microsoft.CodeAnalysis.Host.Mef;
+using Microsoft.CodeAnalysis.LanguageServer;
 using Microsoft.CodeAnalysis.PooledObjects;
 using Microsoft.CodeAnalysis.Shared.Collections;
 using Microsoft.CodeAnalysis.Shared.TestHooks;
-using Microsoft.CodeAnalysis.SolutionCrawler;
+using Microsoft.ServiceHub.Framework;
+using Microsoft.VisualStudio.RpcContracts.DiagnosticManagement;
+using Microsoft.VisualStudio.RpcContracts.Utilities;
+using Microsoft.VisualStudio.Shell.ServiceBroker;
 using Roslyn.Utilities;
+using static Microsoft.ServiceHub.Framework.ServiceBrokerClient;
 
 #pragma warning disable CA1200 // Avoid using cref tags with a prefix
 
 namespace Microsoft.VisualStudio.LanguageServices.Implementation.TaskList;
-
-using ProjectErrorMap = ImmutableDictionary<ProjectId, ImmutableArray<DiagnosticData>>;
-
 /// <summary>
 /// Diagnostic source for warnings and errors reported from explicit build command invocations in Visual Studio.
 /// VS workspaces calls into us when a build is invoked or completed in Visual Studio.
 /// <see cref="ProjectExternalErrorReporter"/> calls into us to clear reported diagnostics or to report new diagnostics during the build.
 /// For each of these callbacks, we create/capture the current <see cref="GetBuildInProgressState()"/> and
 /// schedule updating/processing this state on a serialized <see cref="_taskQueue"/> in the background.
-/// The processing phase de-dupes the diagnostics reported from build and intellisense to ensure that the error list does not contain duplicate diagnostics.
-/// It raises events about diagnostic updates, which eventually trigger the "Build + Intellisense" and "Build only" error list diagnostic
-/// sources to update the reported diagnostics.
 /// </summary>
+[Export(typeof(ExternalErrorDiagnosticUpdateSource))]
```

Comment:
meat of the change in this type.  The change here is to report build diagnostics to the editor's diagnostic brokered service.  Without this, the build diagnostics reported here go into the void.  THis was fine for CPS projects because they are responsible for reporting build errors to the service.

However legacy csproj project system was not updated to do the same, so build diagnostics for legacy csproj went into the void.

This changes our reporter to instead report the build diagnostics to the brokered service.  Long term legacy csproj should be updated to use the brokered service themselves, but requires additional work to enable brokered service usage in C++.